src/RefractDataStructure.cc

```diff
@@ -372,6 +447,29 @@ namespace drafter {
             }
         };
 
+        template<typename X, bool dummy = true>
+        struct InjectNestedTypeInfoSourceMaps {
+            void operator()(const SectionInfo<mson::ValueMember>&, std::vector<snowcrash::SourceMap<typename T::ValueType> >&) {
+            }
+        };
+
+        template<bool dummy>
+        struct InjectNestedTypeInfoSourceMaps<RefractElements, dummy> {
+            void operator()(const SectionInfo<mson::ValueMember>& valueMember, std::vector<snowcrash::SourceMap<typename T::ValueType> >& values) {
```

Comment:
You have `ValueSourceMapCollectionType` in `ElementData` which can be used here.